options.cpp

```diff
@@ -49,6 +49,24 @@ RtSmithOptions::RtSmithOptions()
             return true;
         },
         "Write the generated config to the specified .txtpb file.");
+    registerOption(
+        "--user-p4info", "filePath",
+        [this](const char *arg) {
+            _userP4Info = arg;
+            if (!std::filesystem::exists(_userP4Info.value())) {
+                ::error("%1% does not exist. Please provide a valid file path.",
+                        _userP4Info.value().c_str());
+                return false;
+            }
+            if (_p4InfoFilePath.has_value()) {
```

Comment:
We can put this into the validate function of the options now. We can override this function.